06.wc/wc.rb

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def exec
+  options = parse_options
+  options = { c: true, l: true, w: true } if options.empty?
+  paths = ARGV
+  paths.empty? ? display_word_count_from_stdin(options) : display_word_count_from_files(options, paths)
+end
+
+def parse_options
+  opt = OptionParser.new
+  options = {}
+  opt.on('-c')
+  opt.on('-l')
+  opt.on('-w')
+  opt.parse!(ARGV, into: options)
+  options
+end
+
+def display_word_count_from_stdin(options)
+  lines = $stdin.readlines
+  word_count_map = build_word_count_map(lines)
+  display_word_count(word_count_map, options)
+end
+
+def build_word_count_map(lines)
+  {
+    number_of_lines: count_line(lines),
+    number_of_words: count_word(lines),
+    bytesize: count_bytesize(lines)
+  }
+end
```

Comment:
これは趣味の範囲ですが、

```suggestion
def build_word_count_map(lines, path = '')
  {
    number_of_lines: count_line(lines),
    number_of_words: count_word(lines),
    bytesize: count_bytesize(lines),
    path: path
  }
end
```

として、

```ruby
word_counts << " #{word_count_map[:path]}" unless word_count_map[:path].empty?
```

みたいにすれば、

```ruby
def build_word_count_maps(paths)
  paths.map do |path|
    lines = File.readlines(path)
    build_word_count_map(lines, path)
  end
end
```

と書けそうです。